in_toto/model.go

```diff
@@ -211,6 +222,16 @@ func validateSupplyChainItem(supplyChainItem SupplyChainItem) error {
 	if supplyChainItem.Name == "" {
 		return fmt.Errorf("name cannot be empty")
 	}
+	for _, rule := range supplyChainItem.ExpectedMaterials {
```

Comment:
Previously, we discussed having a helper for iterating and verifying signatures. We also do have something similar in paramsub, though that was messier.

https://github.com/in-toto/in-toto-golang/blob/0e8eb5b8845baffc91cd4b494772b2b77c4ddf0f/in_toto/model.go#L98-L120

https://github.com/in-toto/in-toto-golang/blob/7e0a1a42ec0c1d5d894bd576fbd8fed1cb01da12/in_toto/verifylib.go#L637-L655

Think that's relevant here?

Reply:
Not sure how it is relevant. Can you elaborate?

Reply:
Basically, I’m asking if there should be a verifyArtifactRules that in turn iterates and calls verifyArtifactRule, similar to how it’s setup for signatures etc elsewhere.

Reply:
Done in 0766aa7